optuna/storages/rdb/storage.py

```diff
@@ -326,14 +326,58 @@ def get_all_study_summaries(self):
 
         return study_sumarries
 
-    def create_new_trial_id(self, study_id):
-        # type: (int) -> int
+    def create_new_trial_id(self, study_id, template_trial=None):
+        # type: (int, Optional[structs.FronzenTrial]) -> int
 
         session = self.scoped_session()
 
-        trial = models.TrialModel(study_id=study_id, state=structs.TrialState.RUNNING)
+        if template_trial is None:
+            trial = models.TrialModel(study_id=study_id, state=structs.TrialState.RUNNING)
+        else:
+            # Because only `RUNNING` trials can be updated,
+            # we temporarily set the state of the new trial to `RUNNING`.
+            # After all fields of the trial have been updated,
+            # the state is set to `template_trial.state`.
+            temp_state = structs.TrialState.RUNNING
+
+            trial = models.TrialModel(
+                study_id=study_id,
+                state=temp_state,
+                value=template_trial.value,
+                datetime_start=template_trial.datetime_start,
+                datetime_complete=template_trial.datetime_complete,
+            )
 
         session.add(trial)
+        session.flush()
+
+        if template_trial is not None:
+            for param_name, param_value in template_trial.params.items():
+                if param_name not in template_trial.distributions:
+                    session.rollback()
+                    raise ValueError(
```

Comment:
Can't we move this check to `_validate()`?

Reply:
Because `FronzenTrial._validate()` already checks this case (see [here](https://github.com/pfnet/optuna/pull/464/files/8a800900950a554ef4972afe1c74780e5a7488c3#diff-8293551812da2037a53e2807b3337bceR117)), I simply removed the duplicated check code from `storage.py` (https://github.com/pfnet/optuna/pull/464/commits/dc2942d7061b3b04589d79eea4161df15121862d).